contracts/Vault.vy

```diff
@@ -798,7 +816,9 @@ def _issueSharesForAmount(to: address, amount: uint256) -> uint256:
         # Mint amount of shares based on what the Vault is managing overall
         # NOTE: if sqrt(token.totalSupply()) > 1e39, this could potentially revert
         precisionFactor: uint256 = self.precisionFactor
-        shares = precisionFactor * amount * totalSupply / self._totalAssets() / precisionFactor
+
+        freeFunds: uint256 = self._totalAssets() - self._calculateLockedProfit()
+        shares = precisionFactor * amount * totalSupply / freeFunds / precisionFactor
```

Comment:
What testing do we have to make sure this change doesn't introduce a vulnerability?

Reply:
can we modify the deposit and withdraw test in brownie for the same block to assert we got the exact amount of want tokens on both operations?

Reply:
I tough in tests every blockchain call mines a block, I haven't seen an option to batch transactions.
Will look for how that's done.

Reply:
You can shut off auto-mining